client/packages/common/src/ui/layout/tables/context/hooks/useIsGrouped.ts

```diff
@@ -34,3 +35,20 @@ export const useIsGrouped = (key: keyof GroupByItem): IsGroupedControl => {
 
   return { isGrouped: !!groupByItem?.[key], toggleIsGrouped };
 };
+
+export const useIsGroupedState = (key: keyof GroupByItem): IsGroupedControl => {
```

Comment:
just a copy of the above, without the useTable context call - we pass grouping state in rather than relying on context in the new version